### PR TITLE
Issue 1 threading

### PR DESCRIPTION
**Pull request by raphaelnew**
uploadcare-java v3.0
- Support Uploadcare REST API v0.4
- Fixed Threading problem with HttpClient.
- Updated some of deprecated classes.

### CODE REVIEW

File: src/main/java/com/uploadcare/api/RequestHelper.java [hunk 1]
```diff
@@ -165,62 +197,69 @@ public void remove() {
...
+
+        if (statusCode >= 200 && statusCode < 300) {
+            return;
+        } else if (statusCode == 401 || statusCode == 403) {
+            throw new UploadcareAuthenticationException(
+                    streamToString(response.getEntity().getContent()));
+        } else if (statusCode == 400 || statusCode == 404) {
+            throw new UploadcareInvalidRequestException(
+                    streamToString(response.getEntity().getContent()));
+        } else {
```
Comment (dmitry-mukhin) [inline src/main/java/com/uploadcare/api/RequestHelper.java]: what about 3xx?

File: src/main/java/com/uploadcare/urls/CdnPathBuilder.java [hunk 2]
```diff
@@ -182,7 +226,7 @@ public CdnPathBuilder scaleCropCenter(int width, int height) {
      * Flips the image.
      */
     public CdnPathBuilder flip() {
-        sb.append("/-/effect/flip");
+        sb.append("/-/flip");
```
Comment (dmitry-mukhin) [inline src/main/java/com/uploadcare/urls/CdnPathBuilder.java]: here and below, fix related tests

Comment (raphaelnew) [inline src/main/java/com/uploadcare/urls/CdnPathBuilder.java]: fixed in 9f39adcef2170b7657eada4a7229af1f50cf148d

File: src/main/java/com/uploadcare/api/RequestHelper.java (hunk 1 above)
Comment (raphaelnew) [inline src/main/java/com/uploadcare/api/RequestHelper.java]: added in 4d70cab791579f9bcf4c7170058d443e3a24dfab

**Pull request merged**